Mathlib/Algebra/Category/ContinuousCohomology/Basic.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2025 Richard Hill. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Richard Hill, Andrew Yang
+-/
+import Mathlib.Algebra.Category.ModuleCat.Topology.Homology
+import Mathlib.Algebra.Homology.Embedding.Restriction
+import Mathlib.Algebra.Homology.ShortComplex.HomologicalComplex
+import Mathlib.CategoryTheory.Action.Limits
+import Mathlib.Topology.ContinuousMap.Algebra
+
+/-!
+
+# Continuous cohomology
+
+We define continuous cohomology as the homology of homogeneous cochains.
+We define homogenous cochains as `g`-invariant continuous function in `C(G, C(G,...,C(G, M)))`
+instead of the usual `C(Gⁿ, M)` to allow more general topological groups other than locally compact
+ones. For this to work, we also work in `Action (TopModuleCat R) G`, where the `G` action on `M`
+is only continuous on `M`, and not necessarily continuous in both variables, because the `G` action
+on `C(G, M)` might not be continuous on both variables even if it is on `M`.
+
+## Main definition
+- `ContinuousCohomology.homogeneousCochains`:
+  The functor taking an `R`-linear `G`-representation to the complex of homogeneous cochains.
+- `continuousCohomology`:
+  The functor taking an `R`-linear `G`-representation to its `n`-th continuous homology.
+
+## TODO
+- Show that it coincides with `groupCohomology` for discrete groups.
+- Give the usual description of cochains in terms of `n`-ary functions for locally compact groups.
+- Show that short exact sequences induces long exact sequences in certain scenarios.
+-/
+
+open CategoryTheory ContinuousMap
+
+variable (R G : Type*) [CommRing R] [Group G] [TopologicalSpace R]
+
+namespace ContinuousCohomology
+
+variable [TopologicalSpace G] [IsTopologicalGroup G]
+
+variable {R G} in
+/-- The `G` representation `C(G, rep)` given a representation `rep`.
+The `G` action is defined by `g • f := x ↦ f (g⁻¹ * x)`. -/
```

Comment:
Is this right? It's not what the Lean says as far as I can see (there's no rho(g) mentioned)